File: algorithms/strings/longest-palindromic-substring/typescript/longestPalindromeSubarray.ts
export function longestPalindromeSubarray(arr: number[]): number {
    const n = arr.length;
    if (n === 0) return 0;

    function expand(l: number, r: number): number {
        while (l >= 0 && r < n && arr[l] === arr[r]) { l--; r++; }
        return r - l - 1;
    }

    let maxLen = 1;
    for (let i = 0; i < n; i++) {
        const odd = expand(i, i);
        const even = expand(i, i + 1);
        maxLen = Math.max(maxLen, odd, even);
    }
    return maxLen;
}

File: algorithms/strings/longest-palindromic-substring/csharp/LongestPalindromeSubarray.cs
using System;

public class LongestPalindromeSubarray
{
    public static int Solve(int[] arr)
    {
        int n = arr.Length;
        if (n == 0) return 0;

        int maxLen = 1;
        for (int i = 0; i < n; i++)
        {
            int odd = Expand(arr, i, i);
            int even = Expand(arr, i, i + 1);
            maxLen = Math.Max(maxLen, Math.Max(odd, even));
        }
        return maxLen;
    }

    private static int Expand(int[] arr, int l, int r)
    {
        while (l >= 0 && r < arr.Length && arr[l] == arr[r])
        {
            l--;
            r++;
        }
        return r - l - 1;
    }
}
